Additional SHOW INDEXES/CONSTRAINTS documentation

Hunterness opened the pull request.

Hunterness committed:
Add SHOW INDEXES/CONSTRAINTS to PrettifierParser

Hunterness committed:
Add SHOW INDEXES/CONSTRAINTS to execution plans section

Lojjs commented on cypher/cypher-docs/src/docs/dev/execution-plans.asciidoc:
```diff
@@ -144,6 +144,8 @@ This table comprises all the execution plan operators ordered lexicographically.
 | <<query-plan-let-select-or-anti-semi-apply, LetSelectOrAntiSemiApply>>     | Performs a nested loop. Tests for the absence of a pattern predicate that is combined with other predicates. | | |
 | <<query-plan-let-semi-apply, LetSemiApply>>                                | Performs a nested loop. Tests for the presence of a pattern predicate in queries containing multiple pattern predicates. | | |
 | <<query-plan-limit, Limit>>                                                | Returns the first 'n' rows from the incoming input. | | |
+| <<query-plan-list-constraints, ListConstraints>>                           | List constraints with either verbose of brief output. | Y | |
+| <<query-plan-list-indexes, ListIndexes>>                                   | List indexes with either verbose of brief output. | Y | |
```
Since the first column is the name of the operator, I think we should write `ShowConstraints` and `ShowIndexes` to match the execution plan that is printed out when you run EXPLAIN. The description could be either show or list, even if I prefer show slightly for consistency.

Hunterness replied:
The consistency is that docs use ´list` everywhere :P But I agree that I should update the operator name to the correct one

Lojjs commented on cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/QueryPlanTest.scala:
```diff
@@ -273,6 +273,23 @@ class QueryPlanTest extends DocumentingTestBase with SoftReset {
     )
   }
 
+  @Test def showConstraints() {
```
This is not rendered at all, I think you need to include it in the file operators.asciidoc

Hunterness replied:
yes I do

Lojjs commented on cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/QueryPlanTest.scala:
```diff
@@ -343,6 +360,23 @@ class QueryPlanTest extends DocumentingTestBase with SoftReset {
     )
   }
 
+  @Test def showIndexes() {
```
This is not rendered at all, I think you need to include it in the file operators.asciidoc

Lojjs commented on cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/QueryPlanTest.scala:
```diff
@@ -343,6 +360,23 @@ class QueryPlanTest extends DocumentingTestBase with SoftReset {
+    }
+
+    profileQuery(
+      title = "List indexes",
```
For consistency I would prefer the title to be "Show indexes" (remember to also update the link in the execution plans table)

Hunterness replied:
I think that all show commands have their titles be list, I know the index and constraints do because I checked that when choosing title (It was list before when it was a procedure and was not updated when the section started using the command)

Lojjs replied:
Ok that is true, if you fix the operators names and the includes/rendering, this can be as it is.

Lojjs commented on cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/QueryPlanTest.scala:
```diff
@@ -273,6 +273,23 @@ class QueryPlanTest extends DocumentingTestBase with SoftReset {
+    }
+
+    profileQuery(
+      title = "List constraints",
```
For consistency I would prefer the title to be "Show constraints" (remember to also update the link in the execution plans table)

Lojjs reviewed (changes requested):
Some comments on the rendering and show vs. list

Hunterness committed:
Address review comments

Hunterness committed:
Some additional updates around the list index section

Had some thoughts after the index column table renders
to make it easier to follow

Lojjs commented on cypher/cypher-docs/src/docs/dev/ql/administration/indexes-for-search-performance/index.asciidoc:
```diff
@@ -283,8 +283,11 @@ Listing all indexes can be done with `SHOW INDEXES`, which will produce a table
 |===
 
+[NOTE]
+The old built-in procedures for listing indexes, such as `db.indexes`, work as before and are not affected by the
+<<administration-security-administration-database-indexes, `SHOW INDEXES` privilege>>.
```
Should we mention that `db.indexes` is deprecated here?

Lojjs commented on cypher/cypher-docs/src/docs/dev/ql/administration/indexes-for-search-performance/index.asciidoc:
```diff
@@ -207,7 +207,7 @@ include::../indexes/drop-a-non-existing-index.asciidoc[leveloffset=+1]
 
 === List indexes
```
First in the `Indexes for search performance` chapter there is a menu with a link that is supposed to take you here. That link does work in 4.1 but not in 4.2-preview or in your PR. So it was already broken but I happened to see it now.

Hunterness replied:
I guess it broke when Jessica moved the section from the test to this file, will fix :)

Hunterness committed:
Address review comments

Lojjs reviewed (approved):
Great work, now it looks good and all the link works.

AlexicaWright commented on cypher/cypher-docs/src/docs/dev/ql/administration/indexes-for-search-performance/index.asciidoc:
```diff
@@ -283,8 +284,11 @@ Listing all indexes can be done with `SHOW INDEXES`, which will produce a table
 |===
 
+[NOTE]
+The old deprecated built-in procedures for listing indexes, such as `db.indexes`, work as before and are not affected by the
```
```suggestion
The deprecated built-in procedures for listing indexes, such as `db.indexes`, work as before and are not affected by the
```

AlexicaWright commented on cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ConstraintsTest.scala:
```diff
@@ -96,7 +98,7 @@ class ConstraintsTest extends DocumentingTestBase with SoftReset {
         """include::list-constraints-table-columns.asciidoc[]
           |
-          |The old built-in procedures for listing constraints, such as `db.constraints`, work as before and are not affected by the
+          |Note that the old deprecated built-in procedures for listing constraints, such as `db.constraints`, work as before and are not affected by the
```
```suggestion
          |Note that the deprecated built-in procedures for listing constraints, such as `db.constraints`, work as before and are not affected by the
```
This should be an admonition, for consistency with its equivalent in the `index` section. I'll be happy to fix in a separate PR.

Hunterness replied:
The only reason its not is due to me not succeeding in doing that using this test setup. Feel free to fix in a separate PR (might be interested in looking at it to see your solution 😄)

AlexicaWright commented on cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SchemaIndexTest.scala:
```diff
@@ -145,16 +145,16 @@ class SchemaIndexTest extends DocumentingTestBase with QueryStatisticsTestSuppor
   @Test def list_indexes() {
     prepareAndTestQuery(
-      title = "Options for listing indexes",
+      title = "Example for listing indexes",
```
```suggestion
      title = "Example of listing indexes",
```

AlexicaWright reviewed (commented):
Looks good, just a few comments. Thank you @Hunterness

Hunterness committed:
Apply suggestions from code review

Co-authored-by: AlexicaWright <49636617+AlexicaWright@users.noreply.github.com>

AlexicaWright committed:
added admonition

Hunterness commented on cypher/cypher-docs/src/docs/dev/ql/administration/constraints/index.asciidoc:
```diff
@@ -252,9 +252,70 @@ include::drop-a-constraint.asciidoc[leveloffset=+2]
+|===
+
+[NOTE]
+Note that the deprecated built-in procedures for listing constraints, such as `db.constraints`, work as before and are not affected by the <<administration-security-administration-database-constraints, `SHOW CONSTRAINTS` privilege>>.
```
```suggestion
The deprecated built-in procedures for listing constraints, such as `db.constraints`, work as before and are not affected by the <<administration-security-administration-database-constraints, `SHOW CONSTRAINTS` privilege>>.
```

To make it consistent with the Index one

AlexicaWright replied:
Of course! Thanks for spotting this!

Hunterness replied:
Also since we move the table of columns in here, shouldn't that mean we can remove the columns file?

Hunterness commented on cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/ConstraintsTest.scala:
```diff
@@ -94,12 +94,11 @@ class ConstraintsTest extends DocumentingTestBase with SoftReset {
-          |Note that the old deprecated built-in procedures for listing constraints, such as `db.constraints`, work as before and are not affected by the
-          |<<administration-security-administration-database-constraints, `SHOW CONSTRAINTS` privilege>>.""".stripMargin,
+        """
+          |To list all constraints with the brief output columns, the `SHOW CONSTRAINTS` command can be used.
```
I'd like to do some smaller changes to this example to make it closer to how the index example works (mentioning the filtering and `VERBOSE`). I can update and push the changes since it differs slightly from index and can't be copied over)

AlexicaWright replied:
Sounds good, please do!

Hunterness commented on cypher/cypher-docs/src/docs/dev/ql/administration/constraints/index.asciidoc:
```diff
@@ -252,9 +252,70 @@ include::drop-a-constraint.asciidoc[leveloffset=+2]
 
 include::drop-a-non-existing-constraint.asciidoc[leveloffset=+2]
 
+
```
not sure why we need/want 2 empty lines here

AlexicaWright replied:
No, one should be enough.

Hunterness committed:
Update so index and constraints sections are more consistent

and remove unused files
